### PR TITLE
Redoing Add-Tenant PR

### PR DESCRIPTION
**Pull request by minpat**
This is a redo of PR# 108.  I've removed the edits to Mutation.cs and Index.tsx, as I think the Mutation.cs file had a conflict before and might not be used anymore.  We'll just need to make sure the Add-Tenant.tsx functionality still works with the new TenantMutations.cs file.

### CODE REVIEW

**Comment by minpat** [comment]
Hey @JamesBock - could you please check this when you have a chance?  The mutation works now in terms of referencing the propertyId and has no errors.  However, I found that if you query the tenant list after adding an item, it shows the propertyId as being null.  Here is the query.  Thanks.

`query getTenants {
  tenants {
    nodes {
      id
      name
      residence {
        propertyId
        address {
          line1
        }
      }
    }
  }
}`

**Comment by JamesBock** [comment]
> Hey @JamesBock - could you please check this when you have a chance? The mutation works now in terms of referencing the propertyId and has no errors. However, I found that if you query the tenant list after adding an item, it shows the propertyId as being null. Here is the query. Thanks.
> 
> `query getTenants { tenants { nodes { id name residence { propertyId address { line1 } } } } }`

This is something I need to add to the TenantMutations. The PropertyId is not being assigned from the TenantInput. Will add to my PR

**Comment by minpat** [comment]
Ok - sounds good - thanks.

File: tenant-file/portal-app/src/component/add-tenant.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        properties {
+            nodes {
+                id
+                name
+            }
+        }
+    }
+`;
+
+// Credit to the Formik documentation - this was largely copied from an example there
```
Comment (JamesBock) [inline tenant-file/portal-app/src/component/add-tenant.tsx:33]: You can remove this

File: tenant-file/portal-app/src/component/add-tenant.tsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        error = "Please make sure to enter your 10-digit cell phone number, without hyphens or parentheses. "
+    }
+    return error;
+}
+
+export default () => {
+
+    const [addTenant] = useMutation(ADD_TENANT);
+
+    const { loading, data } = useQuery(GET_PROPERTIES);
```
Comment (JamesBock) [inline tenant-file/portal-app/src/component/add-tenant.tsx:50]: Is this executing for you locally? It is never firing for me.

**Review by JamesBock — CHANGES_REQUESTED** [review]
A few things to clean up. I'm not getting the query to fire so the properties are always null and nothing shows. Let me know if you want to look at this sometime.

File: tenant-file/portal-app/src/component/add-tenant.tsx (hunk 2 above)
Comment (JamesBock) [reply tenant-file/portal-app/src/component/add-tenant.tsx:50]: I actually didn't have a url set for the apollo client. Works once it is set.

**Comment by JamesBock** [comment]
> A few things to clean up. I'm not getting the query to fire so the properties are always null and nothing shows. Let me know if you want to look at this sometime.

After adding the uri for the ApolloProvider and integrating the changes needed on the backend I made in my recent PR, I was able to add a Tenant, create them a new Residence and assign the Residence to a Property all that the add-tenant page. So this is actually pretty close to being ready. Just need to clean up the data mapping.

File: tenant-file/portal-app/src/component/add-tenant.tsx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+import ReactDOM from "react-dom";
+
+import { Formik, Field, Form } from "formik";
+
+import { gql, useMutation, useQuery } from '@apollo/client';
+
+const ADD_TENANT = gql`
+    mutation addingATenant($fullName: String!, $phoneNumber: String!, $street: String!, $city: String!, $zip: String!, $bldgId: ID) {
+
+        createTenant(inputTenant: { name: $fullName, phoneNumber: $phoneNumber, currentResidence: { addressInput: { line1: $street, city: $city, state: "TX", postalCode: $zip }, propertyId: $bldgId } } )
```
Comment (JamesBock) [inline tenant-file/portal-app/src/component/add-tenant.tsx:12]: We need to map the apartment number with Line2 of the address type to be in line with the USPS standard

**Review by JamesBock — APPROVED** [review]
This is working for me with the backend changes from my PR. Unit number is mapping to line2 appropriately.

**Pull request merged**